docs/dater.py

```diff
@@ -0,0 +1,50 @@
+"""
+Sphinx extension to add timestamps to release notes based on Git versions.
+
+Based on https://github.com/jaraco/rst.linker, with thanks to Jason R. Coombs.
+"""
+
+from __future__ import annotations
+
+import datetime as dt
+import re
+import subprocess
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from sphinx.application import Sphinx
+
+DOC_NAME_REGEX = re.compile(r"releasenotes/\d+\.\d+\.\d+")
+VERSION_TITLE_REGEX = re.compile(r"^(\d+\.\d+\.\d+)\n-+\n")
+
+
+def get_date_for(git_version: str) -> dt.datetime | None:
+    cmd = ["git", "log", "-1", "--format=%ai", git_version]
+    try:
+        out = subprocess.check_output(
+            cmd, stderr=subprocess.DEVNULL, text=True, encoding="utf-8"
```

Comment:
Why are `text` and `encoding` used? From testing locally and from the docs, I would think only one is necessary.

https://docs.python.org/3/library/subprocess.html
> If encoding or errors are specified, or text is true, file objects for stdin, stdout and stderr are opened in text mode using the specified encoding and errors or the [io.TextIOWrapper](https://docs.python.org/3/library/io.html#io.TextIOWrapper) default.

Reply:
Funny story - originally, only `text=True` was used. Then later, when [PEP 597](https://peps.python.org/pep-0597/) introduce encoding warnings when the encoding wasn't specified, indicated to include the encoding. Eventually, the need for `encoding=` will be once again unnecessary (python 3.14 maybe), so we'll want to fall back to `text=`, but until then, both are there. Maybe `text=` could be removed in the interim, but I'm kind-of inclined to leave it there as it's the primary signal.